### PR TITLE
Api end points

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this feature me and my partner did the following

- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post.
- Made the API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    params.require(:comment).permit(:text)
+  end
+
+  def set_user_and_post
+    return if request.headers['Authorization'].nil?
+
+    token = request.headers['Authorization'].split.last
+    @user = User.find_by(token:)
+    @post = @user.posts.includes(:comments).find(params[:post_id])
+  end
```
Comment (OyePriscilla) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] I actually commend your going extra mile for adding token as part of the header, howbeit I was unable to post/get comments from the api despite that I added the token generated from the UI. Please look through the screenshot below that I got from the terminal resolve the issue pointed out. 🙏 

Untitled

- [ ] I got the error message screenshot 👇 when I tried to get a comment from the UI.

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Chimoneg27 and @erickma1 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Has API endpoint that returns all post in JSON format ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/comments_controller.rb (hunk 1 above)
Comment (Chimoneg27) [reply app/controllers/api/v1/comments_controller.rb]: @OyePriscilla what should I change

Comment (OyePriscilla) [reply app/controllers/api/v1/comments_controller.rb]: @Chimoneg27 Perhaps you can turn off the CSRF token validation off; I think this [link](https://stackoverflow.com/questions/35181340/rails-cant-verify-csrf-token-authenticity-when-making-a-post-request) can be of help.

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  def create
+    @comment = @post.comments.build(comment_params.merge(user_id: @user.id))
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: @comment.errors, status: :unprocessable_entity
+    end
+  end
+
```
Comment (Munsa1) [inline app/controllers/api/v1/comments_controller.rb:19]: - [ ] Great job making the requested changes by the previous reviewer, how ever a new error has been introduced still blocking the creation of a comment for a user, kindly fix this. This error occurs when you try to assign a value to an attribute that does not exist in the model. 
The error message indicates that the Comment model does not have an attribute named author_id. 

Error 
image

Screencast
[cinnamon-2023-12-16T154217+0200.webm](https://github.com/Chimoneg27/Rails-Blog-App/assets/81932478/8cd8f31d-182e-4c8e-813e-504617be547f)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Made Some Requested Changes :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @Chimoneg27  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Added endpoint that returns all post in JSON format💯
-  Can create comment via an Endpoint👍
-  Correct git flow🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**